src/mongocxx/test/spec/monitoring.cpp

```diff
@@ -242,6 +242,10 @@ void apm_checker::set_command_started(options::apm& apm) {
                                          kvp("command_name", event.command_name()),
                                          kvp("operation_id", event.operation_id()),
                                          kvp("database_name", event.database_name()))));
+
+        // We MUST NOT report a service_id() outside of load-balancing mode:
+        CHECK_FALSE(event.service_id());
```

Comment:
Can you investigate the new test failures on the patch build attached to this PR? I see error logs for the boost polyfill pointing to this file:

https://spruce.mongodb.com/task/cxx_driver_macos_1014_50_compile_and_test_with_shared_libs_extra_alignment_patch_2fdfd62c60370f660c935f82c1f67b7d4b26d9ab_6115b7740305b94d0870d23b_21_08_13_04_14_38/logs?execution=0

Reply:
Oh, thank you for catching this! It looks like CHECK_FALSE() doesn't know what to do with the underlying boost::optional<>, so I've included the suggested header. However, I don't think this is the right /place/ to do it: I don't see any other references to boost in the tests. It probably needs to go into wherever we're including Catch, or perhaps in stdx? The candidates that stand out to me are third_party/catch/include/catch.hpp (but I imagine we're generating that via catch elsewhere? I don't yet see where...), or src/bsoncxx/stdx/optional.hpp. Let's see if this corrects it, and then figure out if it's best to leave it where it is (we are clearly already using boost) or if we need to squirrel it away maybe in one of those locations (or elsewhere).

Reply:
Update: I stashed this in the optional<> polyfill support, where Boost is specifically wrangled.